src/main/java/uk/gov/hmcts/reform/cpo/controllers/CasePaymentOrdersController.java

```diff
@@ -1,7 +1,59 @@
 package uk.gov.hmcts.reform.cpo.controllers;
 
+import io.swagger.annotations.ApiParam;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.cpo.ApplicationParams;
+import uk.gov.hmcts.reform.cpo.domain.CasePaymentOrder;
+import uk.gov.hmcts.reform.cpo.repository.CasePaymentOrderQueryFilter;
+import uk.gov.hmcts.reform.cpo.service.impl.CasePaymentOrdersServiceImpl;
+import uk.gov.hmcts.reform.cpo.validators.annotation.CaseIdsAnnotation;
+
+import org.springframework.validation.annotation.Validated;
+import uk.gov.hmcts.reform.cpo.validators.annotation.IdsAnnotation;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
 
 @RestController
+@Validated
+@RequestMapping(value = "/api")
 public class CasePaymentOrdersController {
+
+    private final CasePaymentOrdersServiceImpl casePaymentOrdersServiceImpl;
+    private final ApplicationParams applicationParams;
+
+    public CasePaymentOrdersController(CasePaymentOrdersServiceImpl casePaymentOrdersServiceImpl, ApplicationParams applicationParams) {
+        this.casePaymentOrdersServiceImpl = casePaymentOrdersServiceImpl;
+        this.applicationParams = applicationParams;
+    }
+
+    @GetMapping(value = "case-payment-orders", produces = {"application/json"})
+    public List<CasePaymentOrder> getCasePaymentOrders(@ApiParam(value = "list of ids")
```

Comment:
After looking into the spec I think you will need to include additional information about the pagination:
"Expected responses:
One of:
success (with  the full set of fields described in Data Model, together with page, page size and total rows available)"

Could we have the swagger codes with error ones, please? For example:
    @ApiOperation(value = "Get payment orders for a case", notes = "Get payment orders for a case")
    @ApiResponses(value = {
        @ApiResponse(code = 200, message = "Payment orders retrieved"),
        @ApiResponse(code = 400, message = "Bad request")
    })

Reply:
Nice Rafa :) . DONE